R/derive_date_vars.R

```diff
@@ -361,6 +361,55 @@ convert_dtc_to_dtm <- function(dtc,
     as_iso_dtm()
 }
 
+#' Convert a Date into a Datetime Object
+#'
+#' Convert a date (datetime, date, or date character) into a Date vector (usually `'--DTM'`).
+#'
+#' @param dt The date to convert.
+#'
+#'   A date or character date is expected in a format like `yyyy-mm-ddThh:mm:ss`.
+#'   Note: you can use [impute_dtc()] function to build a complete datetime.
+#'
+#' @inheritParams convert_dtc_to_dtm
+#'
+#' @author Samia Kabi
+#'
+#' @return A datetime object
+#'
+#' @keywords computation timing
+#'
+#' @export
+#'
+#' @examples
+#' convert_date_to_dtm("2019-07-18T15:25:00")
+#' convert_date_to_dtm(Sys.time())
+#' convert_date_to_dtm(as.Date("2019-07-18"), time_imputation = "23:59:59")
+#' convert_date_to_dtm("2019-07-18", time_imputation = "23:59:59")
+#' convert_date_to_dtm("2019-07-18")
+convert_date_to_dtm <- function(dt,
+                                date_imputation = NULL,
+                                time_imputation = NULL,
```

Comment:
I'm wondering if we should update the default to align with `time_imputation = "00:00:00"` for `convert_dtc_to_dtm()` and `convert_date_to_dtm()`? Because currently when `time_imputation` is not specified by the user, those functions return `NA` for date-only values. I can update `convert_date_to_dtm()` in this pull request, but I'm not sure if I should touch `convert_dtc_to_dtm()` since it might impact other function calls.

Reply:
@Roche-GSK/admiral-devs , we spotted an inconsistency of the default values for `time_imputation`. In `impute_dtc()` and `derive_vars_dtm()`  `"00:00:00"` is used while `NULL` is used in `convert_*_to_dtm()`.

Should we update the default to `"00:00:00"` for all functions?

What do you think?